homework/index.js

```diff
@@ -29,18 +29,68 @@
     });
     return elem;
   }
-
+  //get repos
   function main(url) {
-    fetchJSON(url, (err, repositories) => {
+    fetchJSON(url, (err, data) => {
       const root = document.getElementById('root');
       if (err) {
         createAndAppend('div', root, { text: err.message, class: 'alert-error' });
-        return;
+      } else {
+        //createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+        const header = createAndAppend('header', root, { class: 'header' });
+        const h1 = createAndAppend('h1', header, { text: 'FooCoding Repos', class: 'h1' });
+        const select = createAndAppend('select', header, { class: 'select' });
+        createAndAppend('option', select, { text: 'Choose your favorite repo' });
+        data.forEach(repo => {
+          const name = repo.name;
+          createAndAppend('option', select, { text: name });
+        });
+        const container = createAndAppend('div', root, { class: 'container' });
+        const repoInfo = createAndAppend('div', container, { class: 'left-div' });
+        const contribs = createAndAppend('div', container, { class: 'right-div' });
+        select.onchange = evt => {
+          const selectedRepo = evt.target.value;
+          const repo = data.filter(r => r.name == selectedRepo)[0];
+          console.log(repo);
+          repoInfo.innerHTML = '';
+          contribs.innerHTML = '';
+
+
+
+          const addInfo = (label, value) => {
+            const repoContainer = createAndAppend('div', repoInfo);
+            createAndAppend('span', repoContainer, { text: label });
+            createAndAppend('span', repoContainer, { text: value });
+          };
+          addInfo('Name: ', repo.name);
+          addInfo('Desciption: ', repo.description);
+          addInfo('Number of forks: ', repo.forks);
+          addInfo('Updated: ', repo.updated_at);
+          //or instead of using the addInfo function we can use these lines:
+          // createAndAppend('div', repoInfo, { text: `Descriptions: ${repo.description}` });
+          // createAndAppend('div', repoInfo, { text: `Number of Forks: ${repo.forks}` });
+          // createAndAppend('div', repoInfo, { text: `Updated at:${repo.updated_at}` });
+          const contribsUrl = repo.contributors_url;
+          fetchJSON(contribsUrl, (err, contributors) => {
+            contributors.forEach(contributor => {
```

Comment:
I realised you don't show errors when loading the contributors, can you have a look at showing it for week 2?